sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/DateExpressionsSuite.scala

```diff
@@ -79,30 +79,22 @@ class DateExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
   }
 
   test("datetime function current_date") {
-    val d0 = DateTimeUtils.currentDate(UTC)
-    val cd = CurrentDate(UTC_OPT).eval(EmptyRow).asInstanceOf[Int]
-    val d1 = DateTimeUtils.currentDate(UTC)
-    assert(d0 <= cd && cd <= d1 && d1 - d0 <= 1)
+    assert(intercept[SparkException] {
+      CurrentDate(UTC_OPT).eval(EmptyRow)
+    }.getMessage.contains("Cannot evaluate expression"))
```

Comment:
instead of asserting that the error message contains substrings, could you use `checkError` to compare the expected error class against the actual one? Here's an example:

https://github.com/apache/spark/blob/c045a425bf0c472f164e3ef75a8a2c68d72d61d3/sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/AlterTableAddPartitionSuite.scala#L42-L51